functions/escape.pp

```diff
@@ -0,0 +1,43 @@
+# @summary Escape strings as systemd-escape does.
+# @param input Input string
+# @param path Use path (-p) ornon-path  style escaping.
+# @return String
+# @example Escaping a string
+#   $result = systemd::escape('foo::bar')
+# @example Escaping a path
+#   $result = systemd::escape('/mnt/foobar',true)
+#
+function systemd::escape(String $input, Boolean $path = false) >> String {
```

Comment:
I think we can prohibit empty strings?
```suggestion
function systemd::escape(String[1] $input, Boolean $path = false) >> String {
```

Reply:
Did first but then 
```
systemd-escape ''
```
is not an error... 

Should test it though.

Reply:
sure, but that doesn't make any sense?

Reply:
Though the fact that

```
 systemd-escape -p  ''
-
```
makes we want to avoid the question.